### ISSUE TITLE
How to get list of custom themes in "UXL-Launcher\Themes" directory AND how to handle theme files that are listed in the InstalledCustomThemes.xml file that aren't in the "Themes" directory.

### ISSUE THREAD

**Issue by DrewNaylor** [DrewNaylor/UXL-Launcher#111]
I just thought of this idea. Instead of trying to loop through each of the `.xml` files in the `\Themes` directory, just make an XML document titled, `InstalledCustomThemes.xml`. This file is meant to contain the name of each theme and the theme's filename itself. 

Info: https://stackoverflow.com/questions/19570022/how-do-you-read-multiple-entries-of-xml-through-vb2010 

Maybe I could do something like below:

```xml

<UXL_Launcher_ThemeList>

    <ThemeEntry>
        <Name>(theme name here)</name> <!-- Name to appear in Options window Personalization tab dropdown box for theme selection. Will also be saved to My.Settings. -->
        <File>ThemeNameHere.xml</File> <!-- Will be used to figure out which file in the "Themes" subdirectory is the theme file to apply.  -->
    </ThemeEntry>

    <ThemeEntry>
        <Name>Example Theme 2</name> <!-- Name to appear in Options window Personalization tab dropdown box for theme selection. Will also be saved to My.Settings. -->
        <File>ExampleTheme2.xml</File> <!-- Will be used to figure out which file in the "Themes" subdirectory is the theme file to apply.  -->
    </ThemeEntry>

<UXL_Launcher_ThemeList>

```

If the user wants to add a new custom theme to this folder, they'll have to first add a new entry to this file before it will show up in the Options window under the Personalization tab. If that (or any) theme file is subsequently removed from that folder, ignore that entry in the `InstalledCustomThemes.xml` file and DON'T show it in the dropdown box for theme selection. If the theme that the user chose is deleted from the folder, apply the `Default` theme instead, but keep their `My.Settings.userChosenTheme` choice, then in the Options window, in the theme selector dropdown box, have the text be the value of `My.Settings.userChosenTheme` and the theme description textbox be something like this:

```vbnet

"The file for the theme, " & My.Settings.userChosenTheme & ", has been moved, renamed, or deleted. The theme's original filename is (File xml element from InstalledCustomThemes.xml). If you keep your current theme selection, when the theme file is restored to (current working directory\Themes), it will again be used instead of the Default theme, though this may require an application restart."

```

**Comment by DrewNaylor** [comment]
This can also be used for the built-in themes, though with some changes:

- Instead of specifying the filename for each of the themes, instead use the `My.Resources.ThemeName` format. To figure out whether a file is built-in or custom, maybe something like the code below could be done:

```vbnet

If (File XML element in file that's passed to this code).StartsWith = "My.Resources." Then
    themeType = "builtin"
Else
    themeType = "custom"
End If

```

**Comment by DrewNaylor** [comment]
Getting the list of themes, both custom and built-in, will probably need an array to store the list in memory along with the file itself or something. I'm not sure yet.

**Comment by DrewNaylor** [comment]
This could help:
https://www.techrepublic.com/article/easily-navigate-xml-with-vbnet-and-xpath/
https://web.archive.org/web/20180329165959/https://www.techrepublic.com/article/easily-navigate-xml-with-vbnet-and-xpath/

**Comment by DrewNaylor** [comment]
For now, this is not going to be used. The info in #112 will be used instead as there's not much time left to mess around with the theme engine. 3.1 needs to be released soon so that 3.2 can be released shortly after Office 2019 is available.
